components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/MyAccountApplicationCreationListener.java

```diff
@@ -0,0 +1,193 @@
+        }
+        // Create Role.
+        if (!isExistingRoleName(myAccountId, tenantDomain)) {
+            addImpersonatorRole(myAccountId, tenantDomain);
+        }
+
+        return true;
+    }
+
+    private void addImpersonatorRole(String myAccountId, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        List<Permission> permissions = new ArrayList<>();
+        Permission permission = new Permission(IMPERSONATE_SCOPE_NAME);
+        permissions.add(permission);
+
+        RoleManagementService roleManagementService = ApplicationManagementServiceComponentHolder.getInstance()
+                .getRoleManagementServiceV2();
+        try {
+            roleManagementService.addRole(IMPERSONATE_ROLE_NAME, new ArrayList<>(), new ArrayList<>(), permissions,
+                    "application", myAccountId, tenantDomain);
+        } catch (IdentityRoleManagementException e) {
+            throw new IdentityApplicationManagementException("Error occurred while creating impersonator role.");
+        }
+    }
+
+    private boolean isExistingRoleName(String myAccountId, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        RoleManagementService roleManagementService = ApplicationManagementServiceComponentHolder.getInstance()
+                .getRoleManagementServiceV2();
+        try {
+            return roleManagementService.isExistingRoleName(IMPERSONATE_ROLE_NAME,
+                    "application", myAccountId, tenantDomain);
+        } catch (IdentityRoleManagementException e) {
+            throw new IdentityApplicationManagementException("Error occurred while checking role existence.");
+        }
+    }
+
+    private void addApiResourceToApplication(String myAccountId, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        APIResource apiResource = getImpersontionAPIResource(tenantDomain);
+
+        AuthorizedAPI authorizedAPI = new AuthorizedAPI(
+                myAccountId,
+                apiResource.getId(),
+                RBAC_AUTHORIZATION,
+                apiResource.getScopes(),
+                TENANT
+        );
+        try {
+            authorizedAPIManagementService.addAuthorizedAPI(myAccountId, authorizedAPI, tenantDomain);
+        } catch (IdentityApplicationManagementException e) {
+            throw new IdentityApplicationManagementException("Error occured while adding API resource.");
+        }
+    }
+
+    private boolean isAPIResourceRegistered(String myAccountId, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        try {
+            List<AuthorizedAPI> authorizedAPIs = authorizedAPIManagementService.getAuthorizedAPIs(
+                    myAccountId, tenantDomain);
+            for (AuthorizedAPI authorizedAPI : authorizedAPIs) {
+                if (authorizedAPI.getAPIIdentifier().equals("system:impersonation")) {
```

Comment:
Is this line null safe?

Reply:
Addressed with 82654fc5ad702dedb715d648bde39c8f8b15aa16